prov/util/src/util_buf.c

```diff
@@ -45,6 +45,74 @@ enum {
 };
 
 
+static int ofi_bufpool_region_alloc(struct ofi_bufpool_region *buf_region)
+{
+	int ret;
+	ssize_t page_size;
+	size_t alloc_size;
+	struct ofi_bufpool *pool = buf_region->pool;
+
+	if (pool->attr.flags & OFI_BUFPOOL_HUGEPAGES) {
+		page_size = ofi_get_hugepage_size();
+		if (page_size > 0 && (ssize_t) pool->alloc_size >= page_size) {
+			alloc_size = ofi_get_aligned_size(pool->alloc_size, page_size);
+			ret = ofi_alloc_hugepage_buf((void **) &buf_region->alloc_region,
+					     alloc_size);
+			if (!ret) {
+				buf_region->flags = OFI_BUFPOOL_HUGEPAGES | OFI_BUFPOOL_MMAP;
+				pool->alloc_size = alloc_size;
+				pool->region_size = pool->alloc_size - pool->entry_size;
+				return 0;
+			}
+		}
+		/* If we can't allocate huge pages, fall back to normal
+		 * allocations for all future attempts.
+		 */
+		pool->attr.flags &= (~OFI_BUFPOOL_HUGEPAGES) | OFI_BUFPOOL_MMAP;
```

Comment:
Is this correct? I think `OFI_BUFPOOL_MMAP` need to be added to flag. so should be `|=`

Reply:
Yes of course!